From b15391c1d49745d70999e8b7f5f781dfb8376f82 Mon Sep 17 00:00:00 2001
From: chandrap <prakash1.chandra@intel.com>
Date: Fri, 21 Aug 2015 15:46:51 -0700
Subject: [PATCH 149/441] using common _tistdtypes.h ported from NPCPU code
 changing hwMbox_types.h to use _tistdtypes.h file for NPCPU typesdef data
 types

---
 include/linux/avalanche/generic/_tistdtypes.h | 19 ++++++++++---------
 include/linux/hwMbox_types.h                  | 10 +++-------
 2 files changed, 13 insertions(+), 16 deletions(-)

--- a/include/linux/avalanche/generic/_tistdtypes.h
+++ b/include/linux/avalanche/generic/_tistdtypes.h
@@ -91,11 +91,11 @@
 /**
  * \enum Bool - Boolean Type
  */
-typedef enum 
-{
-    False = 0,
-    True = 1
-} Bool;
+
+typedef bool            Bool;
+
+#define True true
+#define False false
 
 typedef int             Int;    /**< Signed base integer quanity */
 typedef unsigned int    Uns;    /**< This is depricated support only */
@@ -107,15 +107,16 @@ typedef void*           Ptr;    /**< Arb
 
 /* Signed integer definitions (64bit, 32bit, 16bit, 8bit) follow... */ 
 typedef long long       Int64; 
-typedef int             Int32;
+typedef int32_t         Int32;
 typedef short           Int16;
 typedef signed char     Int8; 
 
 /* Unsigned integer definitions (64bit, 32bit, 16bit, 8bit) follow... */ 
-typedef unsigned long long  Uint64; 
-typedef unsigned int        Uint32;
+typedef unsigned long long  Uint64;
+typedef uint32_t            Uint32;
+typedef uint8_t             Uint8;
 typedef unsigned short      Uint16;
-typedef unsigned char       Uint8;
+
 
 /*@}*/
 #endif /* _TI_STD_TYPES */
--- a/include/linux/hwMbox_types.h
+++ b/include/linux/hwMbox_types.h
@@ -67,13 +67,9 @@
 */
 
 
+#include <linux/avalanche/generic/_tistdtypes.h>
 
-
-typedef uint32_t Uint32;
-typedef int32_t  Int32;
-typedef uint8_t  Uint8;
-typedef bool     Bool;
-#define True true
-#define False false
+//#define True true
+//#define False false
 #include <linux/types.h> /* include file with generic typedefs for APP-CPU */
 
